### PR TITLE
[wgsl] Update Goals section of spec

### PR DESCRIPTION
**Pull request by dj2**
This CL updates the goals section with the original language combined
with the discussion from #586 and the weekly meeting.

Issue #586

### CODE REVIEW

File: wgsl/README.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # WebGPU Shading Language Specification
 
+## Design Criteria
+
+There are several constraints driving the development of WGSL. The ones related
+to the language itself are found in the [Goals](https://gpuweb.github.io/gpuweb//wgsl#goals)
+section of the specification. Criteria which guide our development of the language are:
+
+ * The burden for adding constructs to the language is to show why the construct
+   is needed, not to show why it should not be added
```
Comment (damyanp) [inline wgsl/README.md]: I'm having trouble deconstructing this point.  I think I understand it up to the comma (which is what was in #586).  I don't quite get what the part after the comma is trying to add or clarify.

Comment (dj2) [reply wgsl/README.md]: Removed. I was trying to be more clear, but failed, heh.

**Comment by dj2** [comment]
Did anyone else have feedback on this change?

**Comment by damyanp** [comment]
I haven't had a chance to review it, but would like to in the next hour or two.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by damyanp** [comment]
(oops, apologies, I confused this with a different pr!)

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by litherum** [comment]
https://github.com/gpuweb/gpuweb/pull/599 includes the result of our telecon last week.

**Comment by dneto0** [comment]
> #599 includes the result of our telecon last week.

The minutes from last week https://docs.google.com/document/d/15Pi1fYzr5F-aP92mosOLtRL4qLczmVRWQaomfQmGNzo/edit?ts=5e5d1f8e#bookmark=id.f4tachha289j say:

> Resolved: Dan’s initial list of principles from #586 will be accepted. Some of them in the language, some in the README. Dan will send a pull request for this.

Dan posted this PR on the same day.   So #599, which was posted 15hrs ago, should not have tried to address the parts addressed in this PR.

**Comment by litherum** [comment]
> So #599, which was posted 15hrs ago, should not have tried to address the parts addressed in this PR.

Ah, I must have mis-remembered my action item. I'll rebase mine on top of this one.

File: wgsl/index.bs [hunk 2]
```diff
@@ -40,10 +40,14 @@ thead {
...
+ * [SHORTNAME] is as small as possible to represent what is needed
+ * Strong static typing
+ * Directly represents structured control flow
+ * Enables simple and efficient compilation to SPIR-V, Metal Shading Language, and HLSL
+     * No polymorphism, no general pointers, no overloads, etc
+ * Enables simple and efficient conversion from SPIR-V to [SHORTNAME] to SPIR-V while preserving the performance properties of the original shader
  * Constructs are defined as normative references to their SPIR-V counterparts
- * All features in [SHORTNAME] are directly translatable to SPIR-V. (No polymorphism, no general pointers, no overloads, etc)
- * Features and semantics are exactly the ones of SPIR-V
  * Each item in this spec *must* provide the mapping to SPIR-V for the construct
```
Comment (litherum) [inline wgsl/index.bs]: We should probably join these two together into a single item. What's the difference between "constructs" and "items?"

It also seems that they aren't really goals, but instead are a means to achieving the "simple and efficient conversion from SPIR-V." These two probably belong in the README instead.

Comment (damyanp) [reply wgsl/index.bs]: In my mind these two are different to the simple and efficient conversion goal.  The key part here is that the semantics of WGSL are inherited from SPIR-V.  This is why we can have confidence that WGSL works as a shader language and doesn't specify things that are inherently hard or ambiguous across multiple GPU architectures.  This is how WGSL leverages all the work that has been done to ensure this in SPIR-V.

Comment (dj2) [reply wgsl/index.bs]: I'm in agreement with @damyanp here. The semantics of SPIR-V are an important goal for WGSL.

**Comment by dj2** [comment]
ping

**Comment by dj2** [comment]
Closing for now, we can revisit after the FAQ work is done.

**Pull request closed without merging**